src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -711,8 +763,112 @@ ConvSolution InitInvokerFactoryWrwNCHW(const ExecutionContext& ctx,
     using Input2 = internal::CKTransposeInputOp<ND, internal::ConvOperandTag::Output>;
     using Output = internal::CKTransposeOutputOp<ND, internal::ConvOperandTag::Weights>;
 
-    return InitInvokerFactoryNCHW<DeviceOpType, CKArgsType, CastType>(
-        ctx, problem, kernel_id, Input1{}, Input2{}, Output{});
+    assert(problem.IsLayoutDefault());
+
+    ConvSolution result;
```

Comment:
This giant duplication of code can be avoided by checking if the `output_tr_inst.GetConvOperandTag() == internal::ConvOperandTag::Weights` and adding workspace related logic there. See for example line 681 above.

Reply:
The issue was when `CKBWDWeightBufferDescriptor` is created and passed through lambda function but not used in some cases (fwd and bwd) compiler throws warning (error).

Reply:
Let's start a draft PR and point me to it. I'd like to try a couple of ideas. I feel it can be done.

Reply:
ok will do in https://github.com/ROCm/MIOpen/pull/2972